PORT-9395 | Update sonarqube guide to use search relation (#1464)
port-labs/port-docs: docs/build-your-software-catalog/sync-data-to-catalog/code-quality-security/sonarqube/guides/connect-github-pr-with-sonar-analysis.md

<br/><br/>

Now that the <PortTooltip id="blueprint">blueprints</PortTooltip> are related,
we need to assign the relevant SonarQube analysis to each of our pull requests.
This can be done by adding some mapping logic.
Go to your [data sources page](https://app.getport.io/settings/data-sources),
and click on your GitHub Pull Request integration:
<img src='/img/guides/githubPrIntegrationDataSource.png' border='1px' />

<br/><br/>

## Relation Mapping

> There are two ways
> to establish the relationship between a GitHub `pull-request` entity and a SonarQube `analysis` entity:

Reviewer: ```suggestion
We will explore two ways to establish the relationship between a GitHub `pull-request` entity and a SonarQube `analysis` entity:
```

### Matching Common Pull Request Title and Branch

> The first way is to use the `title` and `branch` properties:

Reviewer: ```suggestion
The first approach is to establish a match using the title and branch properties
```

- **Why use `title` and `branch` properties?**
  The `title` property is common to both GitHub pull requests and SonarQube analyses,
  making it a reliable identifier for matching related entities.

>   The `branch` property helps specify the exact code changes,
>   ensuring accurate mapping even if there are pull requests with similar titles on different branches.

Reviewer: ```suggestion
  The `branch` property gives information about the source and destination of the the code changes. Combining the title with the branch offers a robust mapping, ensuring that entities are correctly matched in situations where there are similar pull request titles but on different branches.
```

- Go to your [data sources page](https://app.getport.io/settings/data-sources) and click on the GitHub exporter:
  <img src='/img/guides/githubExporter.png' width='100%' border='1px' />

- Replace the mapping YAML with this content:

  ```yaml showLineNumbers
  resources:

>   - kind: pull-request

Reviewer 2: fix indentation of this code snippet

Author: Done

Author: thank you it was an oversight

          entity:
             mappings:
                identifier: .head.repo.name + '-' + (.number|tostring)
                title: .title
                blueprint: '"githubPullRequest"'
                properties:
                   creator: .user.login
                   assignees: '[.assignees[].login]'
                   reviewers: '[.requested_reviewers[].login]'
                   status: .status
                   closedAt: .closed_at
                   updatedAt: .updated_at
                   mergedAt: .merged_at
                   prNumber: .id
                   link: .html_url
                relations:
                   sonarAnalysis:
                      combinator: '"and"'
                      rules:
                         - property: '"$title"'
                           operator: '"="'
                           value: .title
                         - property: '"branch"'
                           operator: '"="'
                           value: .head.ref

  ```

This configuration uses the `title` and `branch` properties to establish a relationship with SonarQube analysis based on matching properties.

### Matching Pull Request Commit SHA

> The second way is to use the `commitSha` property:

Reviewer: ```suggestion
The second option is to use the `commitSha` property that is common to both the pull request and SonarQube analysis:
```
